src/main/java/nu/studer/gradle/buildscan/teamcity/TeamCityBuildScanPlugin.java

```diff
@@ -22,25 +22,23 @@ public void apply(Project project) {
         if (System.getenv(TEAMCITY_VERSION_ENV) != null && System.getenv(GRADLE_BUILDSCAN_TEAMCITY_PLUGIN_ENV) == null) {
             project.getLogger().quiet(ServiceMessage.of(BUILD_SCAN_SERVICE_MESSAGE_NAME, BUILD_SCAN_SERVICE_STARTED_MESSAGE_ARGUMENT).toString());
 
-            project.getPluginManager().withPlugin(BUILD_SCAN_PLUGIN_ID, appliedPlugin -> {
```

Comment:
Is removing this code safe? I would have kept it and extract the contained code block into a method that can then be invoked for the build scan plugin id and the gradle enterprise plugin id.

Reply:
Also, the code is now order-depenend with regards to plugin application.

Reply:
@guylabs here's how it works:

- For Gradle 6+ , we can rely on the extension being present as it was added by the settings plugin. 
- Otherwise, we have to do the `withPlugin` that it was doing to potentially handle the scan plugin coming later.

So, we should first check for the extension and if it's there  use it. Otherwise, we use `withPlugin` to listen for the build scan plugin.

Reply:
Fixed this.